### ISSUE TITLE
Use of isCellEditable causes entire row to toggle its editability, rather than single cell

### ISSUE THREAD

**Issue by mdrake411** [mui/mui-x#11345]
### Search keywords

[data grid] [isCellEditable]

### Latest version

- [X] I have tested the latest version

### Steps to reproduce

Link to live example: [disable editing of specific cells](https://mui.com/x/react-data-grid/editing/#disable-editing-of-specific-cells-within-a-row)

Steps:
1. Build a grid similar to the one in the documentation demo above
2. Set logic to disable cell under certain conditions

### Current behavior

The result is that the all cells in the row are disabled when the logic condition is met.

### Expected behavior

The row should have cells that can be edited while the desired cell is not editable. This would be in line with the property name 'isCellEditable.'

### Context

I have a app being updated that previously used Material Table. It has a requirement where a certain field should not be editable under certain conditions for a given row, while other fields could still be edited.

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by michelengelen** [comment]
Hey @mdrake411 I can confirm this is a bug. Thanks for raising this!

@cherniavskii I'll add this to the board, so we can have a look.
As it seems the description in the docs is also a bit misleading.

It clearly states that `isCellEditable` can be used to define which individual cells are editable,

> You can use the isCellEditable callback prop to define which individual cells the user can edit in a given row.

 but in the last section it says that the entire row will be disabled when the callback returns `false`.
 
 > In the following demo, only the rows with an even Age value are editable. The editable cells have a green background for better visibility.
 
When this is intentional we should rename the prop to `isRowEditable` and adjust the description as well.

WDYT?

**Comment by mdrake411** [comment]
@michelengelen Thanks for taking a look. I agree about the documentation, and the how the name should probably be `isRowEditable`. Furthermore, there is no way to pass in a field name to indicate which cell to disable.

Is there any kind of work-around to make an individual cell non-editable? In the column definition, there is an `editable` property, but I do not see a way to set it programmatically.

**Comment by mdrake411** [comment]
Additional thoughts... if one result of this bug will be to change the name from `isCellEditable` to `isRowEditable`, then there should be a corresponding way to disable the edit button in an action column with `GridActionsCellItem` elements. Otherwise, It could be potentially confusing for a user to click an edit button, have the row turn into edit mode, and not be able to edit any columns.

**Comment by sebcio-o** [comment]
Just encountered this error in my code 😄 Can I fix this?

**Comment by czerwinskilukasz1** [comment]
@michelengelen , @cherniavskii , some of mdrake411's comments suggest that `isCellEditable` should be replaced with controlling editable at the row level only.
However, I can see a lot of use cases when we might want to block just a single cell from being edited, in particular in cases like:
- the current user does not have permissions to edit this field
- the row represents a deleted/archived entity hence some of the properties should not be altered anymore
- some types of rows should allow editing only a subset of cells

**Comment by mdrake411** [comment]
@czerwinskilukasz1 I did not mean to suggest that "isCellEditable should be replaced with controlling editable at the row level only". I was only making the observation that currently it appears that is what it does, despite its property name.

I agree with your use cases, and I have one similar use case. I need the a row to remain editable while a particular cell in that row is not editable depending on the value of another cell in that same row. I believe that would be more representative of the property name.

**Comment by sebcio-o** [comment]
@michelengelen could you assign me to this ticket? I can fix this

**Comment by michelengelen** [comment]
@sebcio-o you're all set ... feel free to reach out if you need help.

**Comment by kvenkatasivareddy** [comment]
is this issue fixed @Sebcio03 @michelengelen ?

**Comment by michelengelen** [comment]
> is this issue fixed @Sebcio03 @michelengelen ?

Not yet ... at least I haven't seen a PR opened for this.

**Comment by Sebcio03** [comment]
> is this issue fixed @Sebcio03 @michelengelen ?

I'll do it utill monday.

**Comment by Sebcio03** [comment]
Hey is there any chat,irc where I can ask for help?

**Comment by michelengelen** [comment]
> Hey is there any chat,irc where I can ask for help?

Yes, we do have a discord community server: [MUI on Discord](https://discord.com/invite/tewQuNeUyS)

**Comment by Zetjen** [comment]
Hi, is this issue already solved?

**Comment by geh18** [comment]
Any update on this one @Sebcio03 :)

**Comment by Sebcio03** [comment]
@geh18 I don't see issue there anymore, I just checked the live example linked in first comment.

`isCellEditable={(params) => params.row.age % 2 === 0}`
- isCellEditable is indeed ran for each cell.
- `params.row` contains the same data for all cells within same row. 

^^^ That's how you block row using value from one particular cell.

To toggle editability for particular cell you can use `params.field` (column name) and `params.value` (cell data) 
Using above example if we would like to block `age` column if `age % 2 === 0` then the code would look like this:

`isCellEditable={(params) => params.field === 'age' & params.value % 2 === 0}`
Or even
`isCellEditable={(params) => params.field === 'age' & params.row.age % 2 === 0}`

![image](https://github.com/user-attachments/assets/eac2067e-4b62-40c5-8446-650006d09d6d)
